code/Test_definitions/LocationRetrieval.feature

```diff
@@ -0,0 +1,83 @@
+Feature: Get a device location (POST /retrieve)
+
+  @location_retrieval_01_by_phoneNumber_or_ipv4Address_or_networkAccessIdentifier
+  Scenario: Get the location of a device
+    Given a valid device <device> identified by a <deviceIdentierName> <deviceIdentierValue>
+    And a provided maxAge of 80 seconds
+    When a retrieve device location request is made 
+    Then the response status code should be 200
+    And the response body should provide a valid area 
+
+    Examples:
+        | deviceIdentifierName     |  deviceIdentifierValue                 |   
+        | phoneNumber              |  +34666111333                          |   
+        | networkAccessIdentifier  |  123456789@domain.com                  |   
+        | ipv6Address              |  2001:db8:85a3:8d3:1319:8a2e:370:7344  |   
+
+
+  @location_retrieval_02_by_ipv4Address
+  Scenario: Get the location of a device
+    Given a valid device <device> identified by a <publicAddress> <privateAddress> <port>
+    And a provided maxAge of 80 seconds
+    When a retrieve device location request is made 
+    Then the response status code should be 200
+    And the response body should provide a valid area 
+
+    Examples:
+      | publicAddress  | privateAddress  | publicPort  | 
+      | 84.125.93.10   | null            | 59765       | 
+      | 84.125.93.10   | 84.125.93.10    | null        |    
+
+
+  @location_retrieval_03_Invalid_MaxAge
+  Scenario: Input an invalid maxAge value (less than 60)
+    Given a valid device identified with phoneNumber +34666111333
+    When a retrieve request is made with a valid maxAge parameter of 59
+    Then the response status code should be 400
+    And the response body should contain an error message indicating an invalid argument
+
+
+  @location_retrieval_04_Missing_MaxAge

```

Comment:
maxAge is not mandatory in rc2: https://github.com/camaraproject/DeviceLocation/blob/main/code/API_definitions/location-retrieval.yaml
The test should not fail with a 400 error, just continue irrespective of age of known location.